bond_test.go

```diff
@@ -225,5 +238,47 @@ func TestSelectOnlyIDs(t *testing.T) {
 	assert.NotEmpty(t, ids[0])
 }
 
+func TestTransaction(t *testing.T) {
+	tx, err := DB.NewTransaction()
+	assert.NoError(t, err)
+
+	// Should fail because user is a UNIQUE value and we already have a "peter".
+	err = DB.User.Tx(tx).Save(&User{Username: "peter"})
+	assert.Error(t, err)
+
+	// Ok, rolling back.
+	err = tx.Rollback()
+	assert.NoError(t, err)
+
+	// Start again.
+	tx, err = DB.NewTransaction()
+
+	// Attempt to add two new unique values.
+	err = DB.User.Tx(tx).Save(&User{Username: "Joe"})
```

Comment:
That's right, it returns a copy of the store that can be used within the transaction context.